Realm/RLMRealm.h

```diff
@@ -495,6 +496,22 @@ __deprecated_msg("Use `performMigrationForConfiguration:error:`") NS_REFINED_FOR
  */
 + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration error:(NSError **)error;
 
+/**
+ Creates a `RLMThreadHandover` from the passed in objects that can be safely imported on a separate thread.
+
+ Used to pass objects between threads in a thread-safe manner.
+
+ @warning Note that the returned `ThreadHandover` must be imported at most once, and that the current version of the
+ Realm will remain pinned until this package is imported or deallocated.
+
+ @param objects The objects to pass between threads.
+ @return A package that can be imported on a separate thread.
+
+ @see RLMThreadHandover
+ */
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
```

Comment:
I personally think that specifying the `Realm` on export makes it more clear that it is an active operation that must be done on the source thread. It also allows for more clear diagnostics when the user passes multiple objects associated with different realms. Finally, all enables passing zero objects (just opening the same `Realm` on another thread).

I also feel that the throwing, tuple-returning import has the most straightforward control flow. It avoid implicit state changes and always allows error handling. Further, it makes it easy to work with the `Realm` and items since they're declared as local variables rather than properties on a struct.